### PR TITLE
Fix #1730: do not refresh asset list when deleting an asset

### LINKED ISSUE #1730: Avoid reloading all assets after deleting an asset from list
This is a small performance issue that has recently bugged me a lot: a full refresh of the asset list is triggered when deleting an asset from the list. This is sometimes slow, especially on accounts with 100+ assets. 

It's not necessary to refresh the list of assets, especially since we already have a method to filter out deleted assets from the search store. We should fix `onDeleteAssetCompleted` so that it clears both the non-categorized `defaultQueryResultsList` and the `defaultQueryCategorizedResultsLists`.

### PR DESCRIPTION
**Pull request by pmusaraj**
Instead, update the list of categorized results in the search store. Fixes #1730

### CODE REVIEW

**Comment by smutnyleszek** [comment]
@pmusaraj would you like me to review this?

File: jsapp/js/searches.es6 [hunk 1]
```diff
@@ -94,7 +93,22 @@ function SearchContext(opts={}) {
           }
         }
       };
+      var filterOutDeletedAssetFromCategorizedList = () => {
+        let list = this.state.defaultQueryCategorizedResultsLists;
+        if (list != undefined) {
```
Comment (smutnyleszek) [inline jsapp/js/searches.es6]: Why not `!==`?

File: jsapp/js/searches.es6 [hunk 2]
```diff
@@ -94,7 +93,22 @@ function SearchContext(opts={}) {
           }
         }
       };
+      var filterOutDeletedAssetFromCategorizedList = () => {
+        let list = this.state.defaultQueryCategorizedResultsLists;
```
Comment (smutnyleszek) [inline jsapp/js/searches.es6]: Do we have any rules on `const` vs `let`? There is an eslint rule about that: https://eslint.org/docs/rules/prefer-const

File: jsapp/js/searches.es6 [hunk 3]
```diff
@@ -80,7 +80,6 @@ function SearchContext(opts={}) {
     },
     onDeleteAssetCompleted (asset) {
       var filterOutDeletedAsset = ({listName}) => {
-        // TODO: look into why sometimes this.state[listName] is not defined
         if (this.state[listName] != undefined) {
```
Comment (smutnyleszek) [inline jsapp/js/searches.es6]: Do you think removing this `// TODO` should leave an explanation comment?

File: jsapp/js/searches.es6 [hunk 4]
```diff
@@ -94,7 +93,22 @@ function SearchContext(opts={}) {
           }
         }
       };
+      var filterOutDeletedAssetFromCategorizedList = () => {
+        let list = this.state.defaultQueryCategorizedResultsLists;
+        if (list != undefined) {
+          var l = {};
+          for (var category in list) {
```
Comment (smutnyleszek) [inline jsapp/js/searches.es6]: Why `var` not `let` or `const`?

File: jsapp/js/searches.es6 [hunk 5]
```diff
@@ -94,7 +93,22 @@ function SearchContext(opts={}) {
...
+            l[category] = list[category].filter(function(result){
+              return result.uid !== asset.uid;
+            });
+          }
+          let o = {};
+          o.defaultQueryCategorizedResultsLists = l;
```
Comment (smutnyleszek) [inline jsapp/js/searches.es6]: Nitpicking: what is `l` and `o`? I would be happier with longer names ;-)

File: jsapp/js/searches.es6 (hunk 2 above)
Comment (pmusaraj) [reply jsapp/js/searches.es6]: We currently don't. I'd rather not change this now, let's note it for a future formatting/code cleanup update.

File: jsapp/js/searches.es6 (hunk 3 above)
Comment (pmusaraj) [reply jsapp/js/searches.es6]: No need. FYI, the search should get an overhaul soon-ish.

File: jsapp/js/searches.es6 (hunk 5 above)
Comment (pmusaraj) [reply jsapp/js/searches.es6]: `filterOutDeletedAssetFromCategorizedList` closely emulates `filterOutDeletedAsset`, and I kept the `l` and `o` variables. They're basically temporary holders of the list data, post-filtering.

File: jsapp/js/searches.es6 (hunk 3 above)
Comment (smutnyleszek) [reply jsapp/js/searches.es6]: Ok

File: jsapp/js/searches.es6 (hunk 5 above)
Comment (smutnyleszek) [reply jsapp/js/searches.es6]: ok :)

**Comment by smutnyleszek** [comment]
@pmusaraj :ship: :it:

**Pull request merged**